Add the boilerpipe parsing adapted from NUTCH-961

jeremie70 opened the pull request:
I adapted a bit the patch from the [NUTCH-961](https://issues.apache.org/jira/browse/NUTCH-961). This patch add the boilerpipe parser in nutch which improve a lot the quality of parser.

lewismc commented on conf/nutch-default.xml:
```diff
@@ -876,6 +876,19 @@
+
+<property>
+  <name>tika.boilerpipe</name>
+  <value>false</value>
```
Can you provide descriptions of these properties please?

lewismc commented on src/plugin/parse-tika/src/java/org/apache/nutch/parse/tika/BoilerpipeExtractorRepository.java:
```diff
@@ -0,0 +1,62 @@
+import java.lang.ClassLoader;
+import java.lang.InstantiationException;
+import java.util.WeakHashMap;
+import org.apache.commons.logging.Log;
```
Nutch currently uses Slf4j

org.slf4j.Logger
org.slf4j.LoggerFactory

I think!

lewismc commented on src/plugin/parse-tika/src/java/org/apache/nutch/parse/tika/BoilerpipeExtractorRepository.java:
```diff
@@ -0,0 +1,62 @@
+          extractorRepository.put(boilerpipeExtractorName, (BoilerpipeExtractor)extractorClass.newInstance());
+
+        } catch (ClassNotFoundException e) {
+          LOG.error("BoilerpipeExtractor " + boilerpipeExtractorName + " not found!");
```
In slf4j we can better structure the catch
http://www.slf4j.org/faq.html#logging_performance
e.g.
```
logger.debug("The entry is {}.", entry);
```

lewismc commented on src/plugin/parse-tika/src/java/org/apache/nutch/parse/tika/TikaParser.java:
```diff
@@ -109,7 +114,18 @@ public Parse getParse(String url, WebPage page) {
+    ContentHandler domHandler;
+    // Check whether to use Tika's BoilerplateContentHandler
+    if (useBoilerpipe) {
+        LOG.debug("Using Tikas's Boilerpipe with Extractor: " + boilerpipeExtractorName);
```
Can also use more efficient slf4j convention
logger.debug("The entry is {}.", entry);

lewismc commented:
Nice patch. Do you have any test cases in mind?

jeremie70 commented:
Thank you a lot for your help. 

I didn't know that this enhancement of theses logs can improve the performance more than 30 times. It's impressive. :)
If you want test it, you have to give theses properties to nutch-site.xml : 
```
<property>
  <name>http.content.limit</name>
  <value>-1</value>
</property>

<property>
  <name>tika.boilerpipe</name>
  <value>true</value>
</property>

<property>
  <name>plugin.includes</name>
 <value>protocol-http|urlfilter-regex|parse-(tika)|index-(basic|anchor)|urlnormalizer-(pass|regex|basic)|scoring-opic</value>
</property>
```
After a recompile you can try and compare with or without tika.boilerpipe : 

> ./nutch indexchecker URL

chrismattmann commented:
@jeremie70 can you please update this PR? I'll go ahead and get it committed or @lewismc can

jeremie70 commented:
I did it. I forgot something ?

chrismattmann commented:
i think you did forget something since I don't see any changes from you here on this PR since @lewismc comments. Did you make sure to push to your remote so this PR gets updated?

jeremie70 commented:
I'm a new user that is possible that i missed something. 
But i pushed a new commit with descriptions in nutch-default.xml, i modified the system of log and i added a property in the indexchecker to see all the field. It's appear in the "files changed" so i think that i updated the patch. Can you check again @chrismattmann  please ? Check if you see a description below the value of :
```
<property>
  <name>tika.boilerpipe</name>
  <value>false</value>
```

lewismc commented:
@jeremie70 no nothing has been updated. Can you please push your branch (say your branch was origin), something like
```
git push origin my-branch
```
Thanks

jeremie70 commented:
I found the problem, i pushed with the bad email, so github didn't show to other my commit. 
I will resolve this tomorow.
Thank you for your help. ;)

jeremie70 commented:
It would be good now. :)

chrismattmann commented:
thanks @jeremie70 i will review cc @lewismc

chrismattmann commented:
merged thanks @jeremie70

dastergon commented:
In which Nutch 2.x version it will land?

lewismc commented:
@dastergon the very next release which is currently slated for 2.4

dastergon commented:
@lewismc is there any ETA for that version?

lewismc commented:
If you can please help me with https://github.com/apache/nutch/pull/99 it would be greatly appreciated.
This is the only blocker.

dastergon commented:
@lewismc ok, I will try to reproduce it
